src/Surfnet/ServiceProviderDashboard/Domain/Entity/EntityDiff.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/**
+ * Copyright 2022 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\ServiceProviderDashboard\Domain\Entity;
+
+class EntityDiff
+{
+    public function __construct($data, $compareTo)
+    {
+        $this->diff = $this->arrayRecursiveDiff($data, $compareTo);
+    }
+
+    public function getDiff(): array
+    {
+        return $this->diff;
+    }
+
+    /**
+     * Recursive diff algorithm.
+     * Kindly shared by mhitza
+     * See https://stackoverflow.com/questions/3876435/recursive-array-diff
+     *
+     * @SuppressWarnings(PHPMD.ElseExpression)
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     */
+    private function arrayRecursiveDiff($aArray1, $aArray2)
```

Comment:
This class would probably benefit from a unittest, as it is in the Domain

Reply:
You are absolutely right. I was not up to spec with our DoD while writing this code change.. Added the test, and found a big fat missing feature while doing so.